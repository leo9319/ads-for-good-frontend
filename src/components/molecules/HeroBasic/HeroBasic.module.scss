@import '../../../styles/breakpoints';
@import '../../../styles/colours';

.heroFixedWidthContainer {
  .heroFixedWidthContainerCtaSection {
    width: 100%;
    left: 0;
    bottom: var(--spacing-lg);
    z-index: 3000;
  }
  .heroDescription {
    color: $color-neutral-600;
  }
  .heroCTAButton {
    font-size: var(--typography-font-size-3) !important;
    padding: var(--spacing-sm) var(--spacing-lg) !important;
  }
}

@media (max-width: $md-min) {
  .heroFixedWidthContainer {
    .heroIcon {
      width: 40px !important;
      height: 40px !important;
      span {
        width: 24px !important;
        height: 24px !important;
      }
    }
  }
}

@media (max-width: $xs-min) {
  .heroFixedWidthContainer {
    .heroCTAButton {
      width: 95%;
    }
  }
}
