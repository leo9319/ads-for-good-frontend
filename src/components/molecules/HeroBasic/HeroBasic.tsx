import React, { useEffect, useState } from 'react';
import { classNames, getModuleClasses } from '@utils/common/classNames';
import Heading from '@radix-styles/atoms/Heading';
import Text from '@radix-styles/atoms/Text';
import Flex from '@radix-styles/atoms/Flex';
import Button from '@radix-styles/atoms/Button';
import Icons from '@components/atoms/Icons';

import styles from './HeroBasic.module.scss';

export interface HeroBasicProps {
  /**
   * Class name to add more styles
   * @property
   */
  className?: string;
  /**
   * Icon name for the Hero baic
   * @property
   */
  iconName?: string;
  /**
   * Title for the Hero baic
   * @property
   */
  title: string;
  /**
   * Description for the Hero baic
   * @property
   */
  description: string;
  /**
   * Button name for the Hero baic
   * @property
   */
  buttonName?: string;
  /**
   * Optional onClick handler for the CTA.
   * @returns void
   */
  onButtonClick?: () => void;
}
export const HeroBasic = ({
  className,
  iconName,
  title,
  description,
  buttonName,
  onButtonClick,
}: HeroBasicProps) => {
  const classes = getModuleClasses(className?.trim(), styles);

  const [isSticky, setIsSticky] = useState(false);
  const [lastScrollY, setLastScrollY] = useState(0);

  const handleScroll = () => {
    const currentScrollY = window.scrollY;
    // Check if the user is scrolling down and if the button should become sticky
    if (currentScrollY > 200 && currentScrollY > lastScrollY) {
      setIsSticky(true);
    } else if (currentScrollY < lastScrollY) {
      // If the user is scrolling up, remove the sticky position
      if (currentScrollY < 300) {
        setIsSticky(false);
      }
    }

    setLastScrollY(currentScrollY);
  };

  useEffect(() => {
    // Add scroll event listener
    window.addEventListener('scroll', handleScroll);

    // Clean up the event listener when the component is unmounted
    return () => {
      window.removeEventListener('scroll', handleScroll);
    };
  }, [lastScrollY]);

  return (
    <>
      <Flex
        gap="5"
        direction="column"
        wrap="wrap"
        align="center"
        justify="center"
        className={classNames(styles.heroFixedWidthContainer, classes)}
      >
        {iconName ? (
          <Icons
            name={iconName}
            iconHeight="40"
            iconWidth="40"
            backgroundColor="#E86100"
            backgroundSize="60px"
            color="#fff"
            rounded
            className={classNames(styles.heroIcon)}
          ></Icons>
        ) : (
          ''
        )}
        <Flex
          gap="4"
          direction="column"
          wrap="wrap"
          align="center"
          justify="center"
        >
          <Heading
            size={{ initial: '7', lg: '10', md: '7', sm: '7' }}
            weight="bold"
            align="center"
            className={classNames(styles.heroTitle)}
          >
            {title}
          </Heading>
          <Text
            size={{ initial: '3', lg: '4', md: '3' }}
            weight="regular"
            align="center"
            className={classNames(styles.heroDescription)}
          >
            {description}
          </Text>
        </Flex>
        {buttonName ? (
          <Flex
            wrap="wrap"
            justify="center"
            className={classNames(styles.heroFixedWidthContainerCtaSection)}
            style={{
              position: isSticky ? 'fixed' : 'initial',
            }}
          >
            <Button
              text={buttonName}
              size="lg"
              mode="primary"
              className={classNames(styles.heroCTAButton)}
              onClick={onButtonClick}
            ></Button>
          </Flex>
        ) : (
          ''
        )}
      </Flex>
    </>
  );
};

export default HeroBasic;
