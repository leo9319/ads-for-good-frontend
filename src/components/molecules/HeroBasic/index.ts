export { default } from './HeroBasic';
export { HeroBasic } from './HeroBasic';
