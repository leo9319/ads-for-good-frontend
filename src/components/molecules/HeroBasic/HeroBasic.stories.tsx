import React from 'react';
import type { Meta, StoryObj } from '@storybook/react';
import HeroBasic from './HeroBasic';
import { Theme } from '@radix-ui/themes';

const meta = {
  title: 'Library/Molecules/HeroBasic',
  component: HeroBasic,
  decorators: [
    Story => (
      <Theme style={{ padding: '20px 16px', boxSizing: 'border-box' }}>
        <Story />
      </Theme>
    ),
  ],
  parameters: {
    layout: 'fullscreen',
  },
  tags: ['autodocs'],
  argTypes: {
    className: {
      control: {
        type: 'text',
      },
    },
    iconName: {
      control: {
        type: 'text',
      },
    },
    title: {
      control: {
        type: 'text',
      },
    },
    description: {
      control: {
        type: 'text',
      },
    },
    buttonName: {
      control: {
        type: 'text',
      },
    },
  },
} satisfies Meta<typeof HeroBasic>;

export default meta;
type Story = StoryObj<typeof meta>;

export const HeroBasicContent: Story = {
  args: {
    className: '',
    iconName: 'icon-campaign',
    title: 'Together, we drive progress',
    description:
      'In 2023, we helped 16.5+ million of the worldâ€™s most vulnerable children and families overcome poverty.',
    buttonName: 'Donate now',
    onButtonClick: () => {
      console.log('Clicked');
    },
  },
};
