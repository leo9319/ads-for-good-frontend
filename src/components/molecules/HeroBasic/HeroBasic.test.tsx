import React from 'react';
import { render } from '@testing-library/react';
import HeroBasic from './HeroBasic';

describe('HeroBasic', () => {
  it('Should render successfully', () => {
    const { baseElement } = render(<HeroBasic title="" description="" />);
    expect(baseElement).toMatchSnapshot();
  });
});
