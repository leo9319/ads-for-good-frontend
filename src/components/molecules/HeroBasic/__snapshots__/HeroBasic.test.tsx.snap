// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroBasic Should render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-Flex rt-r-fd-column rt-r-ai-center rt-r-jc-center rt-r-fw-wrap rt-r-gap-5 heroFixedWidthContainer"
    >
      <div
        class="rt-Flex rt-r-fd-column rt-r-ai-center rt-r-jc-center rt-r-fw-wrap rt-r-gap-4"
      >
        <h1
          class="rt-Heading rt-r-size-6 rt-r-ta-center headingContainer mocked-class-name typography-font-family-default heroTitle"
          id=""
        />
        <span
          class="rt-Text rt-r-ta-center mocked-class-name typography-font-family-default heroDescription"
          id=""
        />
      </div>
    </div>
  </div>
</body>
`;
